tests/test_key_condition_util.py

```
@@ -0,0 +1,75 @@
+    def test_coerce_filter_status_enum_unchanged(self):
+        ex = SearchExpression(FilterStatus=util_b.FilterStatus.FILTER)
+        _coerce_search_expression_filter_status(ex)
+        self.assertIs(ex.FilterStatus, util_b.FilterStatus.FILTER)
+
+    def test_search_simple_path_rejects_gsi_shape_on_search_status(self):
+        table = Table(
+            table_name="t",
+            endpoint_url="http://localhost:8000",
+            region_name="us-west-2",
+            aws_access_key_id="f",
+            aws_secret_access_key="f",
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "aws_secret_access_key"

(S106)